rlclientlib/ranking_event.h

```diff
@@ -170,4 +172,47 @@ namespace reinforcement_learning {
     bool _action_taken = false;
     unsigned int _outcome_type = 0;
   };
+
+  enum payload_type {
+    CB = 1,
+    CCB = 2,
+    SLATES = 3,
+    OUTCOME_SINGLE = 4
+  };
+
+  class generic_event {
```

Comment:
shouldn't we move this to a separate header since it's part of the v2 protocol?

Reply:
Yes, we can do that.